components/protocol/component.go

```diff
@@ -58,6 +58,7 @@ func provide(c *dig.Container) error {
 		return protocol.New(
 			workerpool.NewGroup("Protocol"),
 			p2pManager,
+			p2pManager.GetLocalPeer(),
```

Comment:
Should we better hand it in when actually issuing a block instead of storing it in the protocol? It is also not the peer that is issuing the block but rather the account.